Sources/Datadog/Core/Persistence/Files/Directory.swift

```diff
@@ -43,6 +43,24 @@ internal struct Directory {
             .contentsOfDirectory(at: url, includingPropertiesForKeys: [.isRegularFileKey, .canonicalPathKey])
             .map { url in File(url: url) }
     }
+
+    /// Deletes all files in this directory.
+    func deleteAllFiles() throws {
+        // Instead of iterating over all files and removing them one by one, we create a temporary
+        // empty directory and replace source directory content with (empty) temporary folder.
+        // This makes the deletion atomic, and is more performant in benchmarks.
+        let temporaryDirectory = try Directory(withSubdirectoryPath: "com.datadoghq/\(UUID().uuidString)")
+        _ = try FileManager.default.replaceItemAt(url, withItemAt: temporaryDirectory.url)
+        try? FileManager.default.removeItem(at: temporaryDirectory.url)
+    }
+
+    /// Moves all files from this directory to `destinationDirectory`.
+    func moveAllFiles(to destinationDirectory: Directory) throws {
+        try files().forEach { file in
+            let destinationFileURL = destinationDirectory.url.appendingPathComponent(file.name)
+            try? FileManager.default.moveItem(at: file.url, to: destinationFileURL)
```

Comment:
i guess there is no real solution to this but assuming this `try` will throw sometimes (similar to that extra `,` problem), maybe we can retry the unsuccessful move operations once again, wdyt?

Reply:
We're actually doing this in Android. I think we retry 3 times.

Reply:
Yup, done ✅. We're now retrying the entire `moveAllFiles()` up to 3 times and retrying each `moveItem(at:to:)` up to 3 times as well.